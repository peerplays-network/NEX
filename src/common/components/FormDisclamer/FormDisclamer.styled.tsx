import { styled } from "../../../ui/src";
import { colors } from "../../../ui/src/colors";

export const Disclamer = styled.p`
   {
    font-size: 12px;
    font-weight: 300;
    text-align: center;
    font: normal normal normal 14px/17px Inter;
    letter-spacing: 0px;
    opacity: 1;
    color: ${colors.textColor};
  }
`;
