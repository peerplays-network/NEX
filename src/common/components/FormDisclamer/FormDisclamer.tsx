import React from "react";

import * as Styled from "./FormDisclamer.styled";

type Props = {
  children: React.ReactNode[];
};

const FormDisclamer = ({ children }: Props): JSX.Element => {
  return <Styled.Disclamer>{children}</Styled.Disclamer>;
};

export default FormDisclamer;
