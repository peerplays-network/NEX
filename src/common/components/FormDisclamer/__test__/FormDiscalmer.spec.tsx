import { render } from "@testing-library/react";
import counterpart from "counterpart";
import Link from "next/link";
import React from "react";

import { FormDisclamer } from "../FormDisclamer";

describe("FormDisclamer component", () => {
  it("renders FormDiscalmet correctly in login page", () => {
    const dummyComponentProps = {
      children: (
        <>
          <span>{counterpart.translate(`pages.login.dont_have_account`)}</span>
          <Link href="/signup">
            {counterpart.translate(`links.create_account`)}
          </Link>
        </>
      ),
    };
    const { asFragment } = render(<FormDisclamer {...dummyComponentProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders FormDiscalmet correctly in sign up page", () => {
    const dummyComponentProps = {
      children: (
        <>
          <span>
            {counterpart.translate(`pages.signUp.already_have_account`)}
          </span>
          <Link href="/login">{counterpart.translate(`buttons.login`)}</Link>
        </>
      ),
    };
    const { asFragment } = render(<FormDisclamer {...dummyComponentProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
