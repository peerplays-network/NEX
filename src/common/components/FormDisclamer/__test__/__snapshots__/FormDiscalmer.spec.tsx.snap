// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormDisclamer component renders FormDiscalmet correctly in login page 1`] = `
<DocumentFragment>
  <p
    class="sc-bcXHqe cKWvWQ"
  >
    <span>
      Donâ€™t have an account? 
    </span>
    <a
      href="/signup"
    >
      Create account
    </a>
  </p>
</DocumentFragment>
`;

exports[`FormDisclamer component renders FormDiscalmet correctly in sign up page 1`] = `
<DocumentFragment>
  <p
    class="sc-bcXHqe cKWvWQ"
  >
    <span>
      Already have a Peerplays account? 
    </span>
    <a
      href="/login"
    >
      Log in
    </a>
  </p>
</DocumentFragment>
`;
